bamboozlegui/Karma
# Implement a minimal web API

Things done:
1. Two minimal API controllers - ItemsController and MessagesController. Their use is showcased in Item page (OnGet) and Inbox page (OnGet).
2. Database modification (overhaul). Hate to say it, but I could not resist and finally connect posts to its user. In fact, making the controllers exposed quite some problems with the way we create posts. From now on, posts will not have user related information, but KarmaUser property instead, through which you can access users properties.
3. Slight but meaningful changes to the way our service classes interact with each other. From now on there is more independence between the services (e.g. Picture service pushed out of ItemRepo). The reason for these changes also stem from making WEB API concisely.
4. Overall I very much believe that the code is much more cleaner.

Our API - very barebones but still a starting point:
1. ItemsController is still not fully implemented as in it does not handle pictures. However, that stays true even to our ItemRepository. Instead, my plan would be to create an API that would handle pictures separately which would promote the principle of single responsibility.
2. MessagesController is still quite barebones and I will need to put some great thought on how they will be served through our API.

How to test:
1. Hate to say it, but we should create the database from scratch once again (Drop-Database -> Add-Migration -> Update-Database)
2. Install [postman](https://www.postman.com/)
3. In case you are not familiar with HTTP protocol, spend 5 minutes trying to understand it as well as introduce yourself to the idea of REST API.
4. Try out different HTTP requests with postman (e.g. GET: localhost:5001/api/items).
5. More complicated API requests maybe a little bit too hard to call, but I am willing to help you out with it.

Happy bug searching.
